aggregator/src/aggregator.rs

```diff
@@ -3734,150 +3789,204 @@ pub fn aggregator_filter<C: Clock>(
         upload_responding,
         warp::cors()
             .allow_any_origin()
-            .allow_method("POST")
+            .allow_method("PUT")
             .allow_header("content-type")
             .max_age(CORS_PREFLIGHT_CACHE_AGE)
             .build(),
         response_time_histogram.clone(),
         "upload",
     );
 
-    let aggregate_routing = warp::path("aggregate");
-    let aggregate_responding = warp::post()
+    let aggregation_jobs_put_responding = warp::put()
+        .and(warp::path!(
+            "tasks" / TaskId / "aggregation_jobs" / AggregationJobId
+        ))
+        .and(warp::header::exact(
+            CONTENT_TYPE.as_str(),
+            AggregationJobInitializeReq::<TimeInterval>::MEDIA_TYPE,
+        ))
         .and(with_cloned_value(Arc::clone(&aggregator)))
         .and(warp::body::bytes())
-        .and(warp::header(CONTENT_TYPE.as_str()))
         .and(warp::header::optional::<String>(DAP_AUTH_HEADER))
         .then(
-            |aggregator: Arc<Aggregator<C>>,
+            |task_id: TaskId,
+             aggregation_job_id: AggregationJobId,
+             aggregator: Arc<Aggregator<C>>,
              body: Bytes,
-             content_type: String,
              auth_token: Option<String>| async move {
-                match content_type.as_str() {
-                    AggregateInitializeReq::<TimeInterval>::MEDIA_TYPE => http::Response::builder()
-                        .header(CONTENT_TYPE, AggregateInitializeResp::MEDIA_TYPE)
-                        .body(aggregator.handle_aggregate_init(&body, auth_token).await?),
-                    AggregateContinueReq::MEDIA_TYPE => http::Response::builder()
-                        .header(CONTENT_TYPE, AggregateContinueResp::MEDIA_TYPE)
-                        .body(
-                            aggregator
-                                .handle_aggregate_continue(&body, auth_token)
-                                .await?,
-                        ),
-                    _ => http::Response::builder()
-                        .status(StatusCode::UNSUPPORTED_MEDIA_TYPE)
-                        .body(Vec::new()),
-                }
-                .map_err(|err| Error::Internal(format!("couldn't produce response: {err}")))
+                http::Response::builder()
+                    .header(CONTENT_TYPE, AggregationJobResp::MEDIA_TYPE)
+                    .body(
+                        aggregator
+                            .handle_aggregate_init(&task_id, &aggregation_job_id, &body, auth_token)
+                            .await?,
+                    )
+                    .map_err(|err| Error::Internal(format!("couldn't produce response: {err}")))
             },
         );
-    let aggregate_endpoint = compose_common_wrappers(
-        aggregate_routing,
-        aggregate_responding,
+
+    let aggregation_jobs_put_endpoint = compose_common_wrappers(
+        path_matching_filter(AGGREGATION_JOB_PATH_REGEX),
```

Comment:
We should chain another filter (`warp::filters::method::put()`, etc.) on any cases where we have identical `path_matching_filter(...)` calls. Checking the method in these routing filters will keep us from double-counting DELETE requests in both the PUT and DELETE metrics.

Reply:
I believe we do: every `_responding` filter starts with a method filter and gets `and()`ed with the path matching filter in `compose_common_wrappers`.

Reply:
We need to move (or copy) the method filter from the responding filter to the router filter, or else requests will enter and leave wrapping filters they shouldn't, only to get a rejection from the responding filter.

Reply:
Doing this broke the CORS stuff because if you put the method in the routing filter, you can send OPTIONS anymore. I'll have to pick this up on Monday. It's remarkably hard to get right.

Reply:
Looking at things closer, I think I now see why the histograms don't get double-counted. (the relevant `map()` that gets added after the wrapped filter is skipped if there's a rejection, so it actually can't double count) I think the only downside left is spurious tracing spans, and we can live with that in the meantime.